Foundation/FileManager.swift

```diff
@@ -1381,6 +1381,30 @@ extension FileManager {
         }
         
         override func nextObject() -> Any? {
+
+            func match(filename: String, to options: DirectoryEnumerationOptions, isDir: Bool) -> (Bool, Bool) {
+                var showFile = true
+                var skipDescendants = false
+
+                if isDir {
+                    if options.contains(.skipsSubdirectoryDescendants) {
+                        skipDescendants = true
+                    }
+#if canImport(Darwin)
+                    if options.contains(.skipsPackageDescendants) && filename.hasSuffix(".app") {
+                        skipDescendants = true
+                    }
+#endif
+                }
+                if options.contains(.skipsHiddenFiles) && (filename[filename._startOfLastPathComponent] == ".") {
```

Comment:
Seems like this could be checked first, letting us skip the `isDir` block if true, since it overrides the previous settings. But not a big deal, since the `skipDescendants` check is not expensive.